### ISSUE TITLE
Unable to migrate between resource pools with different roots

### ISSUE THREAD

**Issue by iBrandyJackson** [hashicorp/terraform-provider-vsphere#1870]
### Community Guidelines

- [X] I have read and agree to the [HashiCorp Community Guidelines ](https://www.hashicorp.com/community-guidelines).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform

v1.4.0

### Terraform Provider

v2.3.1

### VMware vSphere

v7

### Description

Unable to migrate a VM between resource pools with different roots.

### Affected Resources or Data Sources

- `vsphere_virtual_machine`
- `vsphere_resource_pool`

### Terraform Configuration

the plan: 

```
# vsphere_virtual_machine.vm will be updated in-place
  ~ resource "vsphere_virtual_machine" "vm" {
      ~ datastore_id                            = "datastore-19" -> "datastore-44"
      ~ host_system_id                          = "host-14" -> "host-42"
        id                                      = "422e4e24-86c9-9ff0-6985-45f3bb0d20a2"
        name                                    = "vm1"
      ~ resource_pool_id                        = "resgroup-9" -> "resgroup-37"
        tags                                    = []
        ^[[90m# (66 unchanged attributes hidden)

      ~ disk {
          ~ datastore_id     = "datastore-19" -> "datastore-44"
            ^[[90m# (19 unchanged attributes hidden)
        }

      ~ network_interface {
          ~ network_id            = "dvportgroup-25" -> "dvportgroup-49"
            ^[[90m# (9 unchanged attributes hidden)
        }

        ^[[90m# (1 unchanged block hidden)
    }
```

### Debug Output

do not have this currently 

### Error Output

`[ERROR] vertex "vsphere_virtual_machine.vm" error: The object or item referred to could not be found.`

### Expected Behavior

Should be able to migrate a VM between resource pools with different roots.

### Actual Behavior

Migrate fails because the Move task is used instead of Migrate task.

### Steps to Reproduce

-Create VM in resource pool in one compute cluster
-Change VM to a resource pool in another compute cluster
-Apply

### References

closed duplicate here: https://github.com/hashicorp/terraform-provider-vsphere/issues/745

**Comment by iBrandyJackson** [comment]
similar issue in [#733](https://github.com/hashicorp/terraform-provider-vsphere/issues/733) and duplicate of closed issue [#745](https://github.com/hashicorp/terraform-provider-vsphere/issues/745)

**Comment by vasilsatanasov** [comment]
Not able to reproduce the issue wit the given details. Having this setup:
![Screenshot 2023-10-12 at 21 19 54](https://github.com/hashicorp/terraform-provider-vsphere/assets/141020316/bc3c5859-763e-449f-b260-d54a389c9652)
VM on Rp-2 in Cluster 2, host is 10.78.183.126
Migrating the VM on Rp-1 on Cluster 1, host is 10.78.180.112, new storage is the local DS of the host.
The HCL:
```
data "vsphere_network" "network" {
	  name          = "VM Network"
	  datacenter_id = data.vsphere_datacenter.datacenter.id
	}

data "vsphere_datastore" "nfs1" {
  name = "VmGenericProvisioningTestbed_NfsDatastore_WJIbz"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}

data "vsphere_datastore" "vmfs1" {
  name = "StdVmfsDatastore_1078180112"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}

data "vsphere_host" "host1" {
  name = "10.78.183.126"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}

data "vsphere_host" "host2" {
  name = "10.78.180.112"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}


data "vsphere_resource_pool" "rp1" {
  name = "Rp-1"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}

data "vsphere_resource_pool" "rp2" {
  name = "Rp-2"
  datacenter_id = data.vsphere_datacenter.datacenter.id
}

resource "vsphere_virtual_machine" "vm" {
  name             = "vm-1-template-clone"
  resource_pool_id = data.vsphere_resource_pool.rp1.id
  host_system_id = data.vsphere_host.host2.id
  guest_id         = "other3xLinuxGuest"
  network_interface {
    network_id   = data.vsphere_network.network.id
  }
  datastore_id     = data.vsphere_datastore.vmfs1.id
  num_cpus = 2
  memory   = 1024

  wait_for_guest_ip_timeout = 0
  wait_for_guest_net_timeout = 0

  disk {
    label            = "disk0"
    size             = 1
    }
}
```
$terraform plan output:
```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # vsphere_virtual_machine.vm will be updated in-place
  ~ resource "vsphere_virtual_machine" "vm" {
      ~ datastore_id                            = "datastore-40" -> "datastore-36"
      ~ host_system_id                          = "host-26" -> "host-33"
        id                                      = "4223464b-a2c9-dbc2-f8f1-646083f1274a"
        name                                    = "vm-1-template-clone"
      ~ resource_pool_id                        = "resgroup-76" -> "resgroup-57"
        tags                                    = []
        # (64 unchanged attributes hidden)

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

```

$terraform apply output:
```

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # vsphere_virtual_machine.vm will be updated in-place
  ~ resource "vsphere_virtual_machine" "vm" {
      ~ datastore_id                            = "datastore-40" -> "datastore-36"
      ~ host_system_id                          = "host-26" -> "host-33"
        id                                      = "4223464b-a2c9-dbc2-f8f1-646083f1274a"
        name                                    = "vm-1-template-clone"
      ~ resource_pool_id                        = "resgroup-76" -> "resgroup-57"
        tags                                    = []
        # (64 unchanged attributes hidden)

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

vsphere_virtual_machine.vm: Modifying... [id=4223464b-a2c9-dbc2-f8f1-646083f1274a]
vsphere_virtual_machine.vm: Still modifying... [id=4223464b-a2c9-dbc2-f8f1-646083f1274a, 10s elapsed]
vsphere_virtual_machine.vm: Modifications complete after 14s [id=4223464b-a2c9-dbc2-f8f1-646083f1274a]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

Closing the issue since I am not able unable to reproduce it
